cimr/processor/processor_prompt.py

```diff
@@ -11,25 +19,88 @@
 
 from .tad import Tadpole
 
+from .yamler import Yamler
+from .yamler import load_yaml
 
-def processor_cli(args):
-    
-    data_types = {'gwas', 'eqtl'}
+from ..defaults import DATA_TYPES
+
+
+def check_type(types):
+    """Check data types"""
+    if types in DATA_TYPES:
+        return True
+    else:
+        logging.error(f' data type is not recognized.')
+        return False
 
-    outdir = args.outdir 
-    outdir.mkdir(exist_ok=True)
-    logging.info(f' directory {str(outdir)} will be used for cimr jobs.')
-    outfile = str(outdir) + '/' + str(args.out)
-    logging.info(f' file prefix {str(args.out)} will be used for output files')
 
-    data_type = args.data_type
+def grow_tadpoles(args):
+    """Standard processing of a new tad annotation file."""
+    tads = Tadpole(
+        file_name=args.file_name,
+        study_id=args.study_id,
+        pub_id=args.pub_id, 
+        species=args.species, 
+        cell_type=args.cell_type,
+        data_type=args.data_type,
+        protocol=args.protocol
+    )
+    tads.read_file()
+    tads.write_file()
+
+
+def convert_yaml(yaml_file):
+    """Convert yaml parameters to cimr arguments"""
+
+    try:
+        yaml_file = pathlib.Path(yaml_file)
+        yaml_file_path = yaml_file.resolve(strict=True)
+        logging.info(f' processing {yaml_file_path}')
+        yaml_data = load_yaml(yaml_file)
+        y = Yamler(yaml_data)
+        y.check_data_file()
+        data_type = y.data_type
+        file_name = pathlib.Path(y.outfile_path)
+        outdir = pathlib.Path('processed_data/' + str(data_type))
+        pathlib.Path('processed_data/').mkdir(exist_ok=True)
+        out_path = y.outfile_path.replace('submitted', 'processed')
+        out_path = out_path
+
+        return data_type, file_name, outdir, out_path
+
+    except FileNotFoundError:
+        logging.info(f' {yaml_file_path} is not accesible')
+        sys.exit(1)
+
+
+def processor_cli(args):
+    """cimr processor subprocess cli"""
 
     if args.process:
-        if data_type in data_types:
+
+        if args.yaml_file:
+            data_type, file_name, outdir, out_path = convert_yaml(args.yaml_file)
+        else:
+            data_type = args.data_type
+            file_name = args.file_name
+            outdir = args.outdir
+            out_path = str(outdir) + '/' + str(args.out)
             
-            if args.file_name is not None:
-                outfile = outfile + '_' + data_type + '.txt.gz'
-                infile = Infiler(data_type, args.file_name, args.genome_build, args.update_rsid, outfile)
+        if check_type(data_type):
+
+            outdir.mkdir(exist_ok=True)
+            logging.info(f' output directory: {str(outdir)}')
+
+            if file_name:
+                outfile = out_path + '.txt.gz'
```

Comment:
In my test cases, the output files in `processed_data/` has the suffix of ".txt.gz.txt.gz`. This is probably because `Yamler` class adds `.txt.gz` suffix too.